### PR TITLE
IGNITE-20296 Make a network message handler that would be able to accept ClusterNode as a sender

### PR DESCRIPTION
**Pull request by rpuch**
https://issues.apache.org/jira/browse/IGNITE-20296

### CODE REVIEW

File: modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/framework/ClusterServiceFactory.java [hunk 1]
```diff
@@ -62,6 +64,13 @@ public class ClusterServiceFactory {
         this.allNodes = allNodes;
     }
 
+    private ClusterNode nodeByName(String name) {
+        return nodeByName.computeIfAbsent(
```
[the remark this answers is not in the document]
Comment (rpuch) [reply modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/framework/ClusterServiceFactory.java:68]: No, same as `messagingServicesByNode` and `topologyServicesByNode`. This is a test harness, this should not create problems.

File: modules/file-transfer/src/test/java/org/apache/ignite/internal/network/file/TestTopologyService.java [hunk 2]
```diff
@@ -67,7 +67,7 @@ public void fairAppearedEvent(ClusterNode member) {
      *
      * @param member Member.
      */
-    public void fairDisappearedEvent(ClusterNode member) {
```
[the remark this answers is not in the document]
Comment (rpuch) [reply modules/file-transfer/src/test/java/org/apache/ignite/internal/network/file/TestTopologyService.java:70]: I encountered the term 'to fire an event' many times, but I don't know what 'to fair an event' might mean :) It seems to me that it was just a typo, so I took a liberty to fix it.

[the remark this answers is not in the document]
Comment (tkalkirill) [reply modules/file-transfer/src/test/java/org/apache/ignite/internal/network/file/TestTopologyService.java:70]: Maybe then the methods above should also be renamed?

File: modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/framework/ClusterServiceFactory.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (tkalkirill) [reply modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/framework/ClusterServiceFactory.java:68]: My mistake was not noticing that this class is in "test".

File: modules/network/src/main/java/org/apache/ignite/internal/network/netty/ConnectionManager.java [hunk 3]
```diff
@@ -109,6 +110,12 @@ public class ConnectionManager implements ChannelCreationListener {
 
     private volatile @Nullable ClusterNode localNode;
 
+    /**
+     * Released when local node is set; attempts to initiate a connection to this node from the outside will wait
+     * till it's released.
+     */
+    private final CountDownLatch localNodeSet = new CountDownLatch(1);
```
[the remark this answers is not in the document]
Comment (rpuch) [reply modules/network/src/main/java/org/apache/ignite/internal/network/netty/ConnectionManager.java]: Switched to the future

**Review by tkalkirill — APPROVED** [review]

**Pull request merged**